### ISSUE TITLE
[BUG] run-script with workspaces does not short-circuit on script error

### ISSUE THREAD

**Issue by johndiiorio** [npm/cli#3806]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

`run-script` with the `--workspaces` flag will run all workspace scripts, regardless if one script exits with exit code 1. Please see the "Steps to Reproduce" section.

### Expected Behavior

I would expect that `run-script` would short-circuit and not run the other scripts if one fails. Alternatively, a new flag (e.g. `--short-circuit`) could be introduced to `run-script` in order to maintain backwards compatibility.

### Steps To Reproduce

Consider the following scenario:
- A root `package.json` file has `"workspaces": ["a", "b", "c"]`
- `a/package.json` has the script `"build": "exit 1"`
- Both `b/package.json` and `c/package.json` have the the script `"build": "exit 0"`

Run `npm run build --workspaces` and note that all scripts are run.

### Environment

- OS: Windows 10
- Node: 16.10.0
- npm: 7.24.0

**Comment by john-landgrave** [comment]
Big upvote on this one. This is kind of a dealbreaker when building a monorepo in CI? If we just execute a `npm run --workspaces package` (or whatever your build/bundle command is) a package can fail and the CI build will still succeed because the exit code comes out 0.

FWIW, this is also the request in this [StackOverflow question](https://stackoverflow.com/questions/71300870/npm-workspace-command-does-not-stop-executing-when-command-fails-for-a-workspace).

**Comment by johndiiorio** [comment]
@john-landgrave I agree, this bug makes it impractical to build a monorepo in a CI with just `npm`. One possible workaround is to create a script (and add it to your root `package.json`) that reads the `package.json` `workspaces` array, then individually invokes `npm run build --workspace <workspace>` and checks the exit code before proceeding. That shouldn't be necessary though, the behavior should be built-in.

**Comment by darcyclarke** [comment]
@johndiiorio Thanks for taking the time to share your idea! New ideas are always appreciated and are better suited for [our RFC repo](https://github.com/npm/rfcs) instead since that's the right place in order to get more attention from the rest of the team and the community.

I've transferred the issue to that repo for future discussion
